pkg/node/node.go

```diff
@@ -1004,18 +1004,44 @@ func NewBee(
 
 	stakingContract := staking.New(overlayEthAddress, stakingContractAddress, abiutil.MustParseABI(chainCfg.StakingABI), bzzTokenAddress, transactionService, common.BytesToHash(nonce), o.TrxDebugMode, uint8(o.ReserveCapacityDoubling))
 
-	if chainEnabled && changedOverlay {
-		stake, err := stakingContract.GetPotentialStake(ctx)
+	if chainEnabled {
+
+		if changedOverlay {
+			stake, err := stakingContract.GetPotentialStake(ctx)
+			if err != nil {
+				return nil, err
+			}
+			if stake.Cmp(big.NewInt(0)) > 0 {
+				logger.Debug("changing overlay address in staking contract")
+				tx, err := stakingContract.ChangeStakeOverlay(ctx, common.BytesToHash(nonce))
+				if err != nil {
+					return nil, fmt.Errorf("cannot change staking overlay address: %v", err.Error())
+				}
+				logger.Info("overlay address changed in staking contract", "transaction", tx)
+			}
+		}
+
+		// make sure that the staking contract has the up to date height
+		tx, updated, err := stakingContract.UpdateHeight(ctx)
 		if err != nil {
-			return nil, errors.New("getting stake balance")
+			return nil, err
+		}
+		if updated {
+			logger.Info("updated new reserve capacity doubling height in the staking contract", "transaction", tx, "new_height", o.ReserveCapacityDoubling)
 		}
-		if stake.Cmp(big.NewInt(0)) > 0 {
-			logger.Debug("changing overlay address in staking contract")
-			tx, err := stakingContract.ChangeStakeOverlay(ctx, common.BytesToHash(nonce))
+
+		if o.ReserveCapacityDoubling > 0 {
+			stake, err := stakingContract.GetPotentialStake(ctx)
```

Comment:
Can we call GetPotentialStake only once , before the two if statements?

Reply:
In a real life scenario, it's very unlikely that we will call get stake twice here